serve/src/br/toc_parser.rs

```diff
@@ -115,48 +155,91 @@ fn parse_toc_header(line_number: usize, line: &str) -> Result<(String, String),
-            // first header
-            if let Some((line_number, line)) = it.next() {
-                let (key, value) = parse_toc_header(line_number, line)?;
-                fill_toc_header(&mut toc, key, value);
-            } else {
-                return Err(TocParseError::MissingTocHeader { line_number });
-            }
 
-            // another header
-            if let Some((line_number, line)) = it.next() {
+        match toc_item_state {
+            TocItemParseState::WaitingTocLine => {
+                if !line.starts_with("- ") {
+                    return Err(TocParseError::ExpectedTocLine { line_number });
+                }
+
+                let (title, path) = parse_toc_line(line_number, line)?;
+                toc.title = title;
+                if let Some(path) = path {
+                    // considering path after the colon
+                    toc.headers.push(("path".to_owned(), path));
+                }
+
+                // checking the future line
+                while let Some((_, future_line)) = iter.peek() {
+                    if future_line.trim().is_empty() {
+                        iter.next();
+                        continue;
+                    }
+                    toc_item_state = next_toc_item_state(future_line, toc_item_state);
+                    if toc_item_state == TocItemParseState::WaitingTocLine {
+                        toc_items.push(toc); // If toc line is always single, it will work
+                        toc = PToc::default()
+                    }
+                    break;
+                }
+            }
+            TocItemParseState::WaitingTocHeader => {
+                if !line.contains(":") {
+                    return Err(TocParseError::MissingTocHeaderColon { line_number });
+                }
                 let (key, value) = parse_toc_header(line_number, line)?;
-                fill_toc_header(&mut toc, key, value);
-            } else {
-                return Err(TocParseError::MissingTocHeader { line_number });
+                toc.headers.push((key, value));
+
+                // checking the future line
+                while let Some((n, future_line)) = iter.peek() {
+                    if future_line.trim().is_empty() {
+                        iter.next();
+                        continue;
+                    }
+                    toc_item_state = next_toc_item_state(future_line, toc_item_state);
+                    if toc_item_state == TocItemParseState::WaitingTocLine {
+                        toc_items.push(toc); // If toc line is always single, it will work
+                        toc = PToc::default()
+                    }
+                    break;
+                }
             }
-            toc_siblings.push(toc);
-        } else {
         }
     }
-    Ok(toc_siblings)
+
+    // if the title of the last item is null it will not include
+    if !toc.title.is_empty() {
```

Comment:
We should check the `headers` also  
```suggestion
    if !toc.title.is_empty() && !headers.is_empty() {
```